src/main/java/uk/gov/di/ipv/cri/common/library/domain/AuditEvent.java

```diff
@@ -43,4 +52,16 @@ public long getTimestamp() {
     public String getEvent() {
         return event;
     }
+
+    public Map<String, String> getRestricted() {
+        return Collections.unmodifiableMap(restricted);
+    }
+
+    public boolean addRestricted(String key, String value) {
+        if (key != null && !key.isBlank() && value != null && !value.isBlank()) {
```

Comment:
If nulls or empty strings are being provided to this method it would be in exceptional circumstances 

Therefore, we can change the return type of this method to `void` and simply mandate that the key/value are specified 
e.g.
if (StringUtils.isBlank(key)) { throw new IllegalArgumentException("key must not be blank"); }

The return type of this method isn't being consumed in the AuditService either
At present, it a null or empty key/value was provided no one would know until it has been sent over to the txma team

Alternatively...

The addRestricted method could be removed in it's entirety and a new setRestricted (mutator method) could be introduced so that the AuditEvent remains as a basic POJO & any input validation of the keys/values happens in the `AuditService`. Given the current makeup of the AuditEvent class this would be my preference